packages/generator/src/service/index-file.ts

```diff
@@ -6,23 +6,26 @@ import {
   StructureKind
 } from 'ts-morph';
 import { VdmServiceMetadata } from '../vdm-types';
+import { hasBatchRequest } from '../generator-utils';
 
 export function indexFile(service: VdmServiceMetadata): SourceFileStructure {
+  const basicStatements = [
+    ...service.entities.map(entity => exportStatement(entity.className)),
+    ...service.entities.map(entity =>
+      exportStatement(`${entity.className}RequestBuilder`)
+    ),
+    ...service.complexTypes.map(complexType =>
+      exportStatement(complexType.typeName)
+    ),
+    ...(service.functionImports && service.functionImports.length
+      ? [exportStatement('function-imports')]
+      : [])
+  ];
   return {
     kind: StructureKind.SourceFile,
-    statements: [
-      ...service.entities.map(entity => exportStatement(entity.className)),
-      ...service.entities.map(entity =>
-        exportStatement(`${entity.className}RequestBuilder`)
-      ),
-      ...service.complexTypes.map(complexType =>
-        exportStatement(complexType.typeName)
-      ),
-      ...(service.functionImports && service.functionImports.length
-        ? [exportStatement('function-imports')]
-        : []),
-      exportStatement('BatchRequest')
-    ]
+    statements: hasBatchRequest(service)
+      ? basicStatements.concat(exportStatement('BatchRequest'))
```

Comment:
Minor: I prefer the spread  [...basicStatements,exportStatement('BatchRequest')]  instead of concat.

Reply:
+1